docker/build.sh

```diff
@@ -64,62 +99,94 @@ function image_variant() {
   fi
 }
 
+function is_buildx() {
+    [[ ${DOCKER_BUILDKIT} -eq 1 ]] \
+      && [[ "${DOCKER_CLI_EXPERIMENTAL}" == 'enabled' ]] \
+      && docker buildx version
+}
+
+function build_with_opts() {
+    local dockerfile=$1; shift
+    local build_stages
+    build_stages="$(grep -Ei '^FROM\s+.*\s+AS\s+.*$' "${dockerfile}" | tr '[:upper:]' '[:lower:]' | awk -F' as ' '{print $2}')"
+
+    if [[ ${build_stages} == '' ]]; then
+        # normal build
+        docker build "$@"
+    else
+        # multistage build handling
```

Comment:
:tada: for buildkit and balena-secrets compat
do you have an example for such a multi-stage build?

Reply:
@robertgzr https://github.com/balenaci/e2e/pull/1